### PR TITLE
Return a 503 if something containing a query string reaches the mirrors

### PR DESCRIPTION
**Pull request by rjbate**
From https://gov-uk.atlassian.net/wiki/display/PLOPS/2016-05-09+Carrenza+disk+problems+-+S1?focusedTaskId=1)

This is a stopgap measure until we do more work on the mirrors.

### CODE REVIEW

File: hieradata/common.yaml [hunk 1]
```diff
@@ -98,7 +98,18 @@ nginx::nginx_locations:
     location: '= /error/503.html'
     location_custom_cfg:
       - 'root /srv/mirror_data'
-
+  'query-string-assets':
+    vhost: 'assets-origin'
```
Comment (surminus) [inline hieradata/common.yaml]: I wonder if we need this for assets at all, maybe @rjw1 knows the answer

File: hieradata/common.yaml [hunk 2]
```diff
@@ -98,7 +98,18 @@ nginx::nginx_locations:
...
+    priority: 900
+    location: ~ \?+
+    location_custom_cfg:
+      - 'return 503 /error/503.html'
+  'query-string-www':
+    vhost: 'www-origin'
+    priority: 900
+    location: ~ \?+
```
Comment (surminus) [inline hieradata/common.yaml]: I suspect this should be a string

Comment (rjbate) [inline hieradata/common.yaml]: I share your suspicion.

File: hieradata/common.yaml (hunk 1 above)
Comment (rjbate) [inline hieradata/common.yaml]: I asked, and the answer was yes, although we hope nothing comes through to assets with a query string in the first place.

**Comment by surminus** [comment]
:+1: I think this looks OK. To test we should just deploy it to a single mirror.

**Comment by rjbate** [comment]
Tested, seems ok locally.

**Comment by surminus** [comment]
:+1: Cool, merge and then we can deploy to one of the mirrors and browse to the URL directly to test.

**Comment by deanwilson** [comment]
Merge when green. On Monday morning.

**Pull request merged**